### ISSUE TITLE
Vendored Botocore Requests

### ISSUE THREAD

**Issue by mikegrima** [pynamodb/PynamoDB#558]
Hello All!

Botocore has begun un-vendoring the Requests library, and replacing it with urllib3: https://github.com/boto/botocore/pull/1495.

PynamoDB is making use of the vendored botocore Requests library here: https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/connection/base.py#L19-L20

As such, this results in breakage when unit testing with the latest version of moto (https://github.com/spulec/moto), which addresses this change.

Is there a major level of effort required to swap out the vendored requests? I would be happy to submit a PR, but I don't have a ton of knowledge on the lower-level logic of PynamoDB. I would imagine that this would will need to be done, since botocore is migrating away from making use of the vendored Requests library.

**Comment by mikegrima** [comment]
Submitted a PR that should hopefully address this properly at: #559.

**Comment by kyleknap** [comment]
I agree with depending directly on `requests` instead of the vendored version of `requests`. The fact that we vendor requests is an implementation detail and it is likely that we remove it in future versions of `botocore`, since we are relying on `urllib3` now.

**Comment by mikegrima** [comment]
@kyleknap Is on the AWS SDK team BTW.

**Comment by asottile** [comment]
additionally, the vendored copy in botocore is vulnerable to CVE 2018-18074

**Comment by dmulter** [comment]
Note that I'm getting a failure using the latest moto, but it's not the exact problem described here. The fix suggested above did not work for me. All my tests pass using moto 1.3.6, but the new one results in failures for mocked DynamoDB tests.

An item is added to a table with no failure, but a subsequent read doesn't return the item data. I'm waiting to see if the fix for this resolves my issue or not. If it doesn't, I'll get more detail and maybe make another issue. Not sure if it's a moto or pynamodb problem, but clearly the tests that always worked are now failing in simple ways with moto 1.3.7.

**Comment by Jamim** [comment]
Related PR: boto/botocore#1817
